src/app/reporting/Engine.cpp

```diff
@@ -336,6 +377,78 @@ void Engine::Run()
         mCurReadHandlerIdx = (mCurReadHandlerIdx + 1) % CHIP_IM_MAX_NUM_READ_HANDLER;
         readHandler        = imEngine->mReadHandlers + mCurReadHandlerIdx;
     }
+    for (auto & path : mDirtyPathList)
+    {
+        path.ClearDirty();
+    }
+    return;
+}
+
+void Engine::SetDirty(ClusterInfo & aClusterInfo)
+{
+    InteractionModelEngine * imEngine = InteractionModelEngine::GetInstance();
+#if !CHIP_SYSTEM_CONFIG_NO_LOCKING
+    ScopedLock lock(*this);
```

Comment:
similar  to question above: why are locks needed? I believe we are moving to a model where the locking is app defined and we assume a single event loop from CHIP perspective.

What is the threading that makes life difficult here? Why is this needed for SetDirty?

Reply:
ok, removed the lock